### ISSUE TITLE
autoscaling.AutoscalingGroup does not implement IGrantable

### ISSUE THREAD

**Issue by abelmokadem** [aws/aws-cdk#4554]
I'm missing `IGrantable` on AutoscalingGroup. The role property is there but the interface `IGrantable` is not implemented.

### Use Case

To make my `eks` life easier.

### Other

* [X] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

**Comment by rix0rrr** [comment]
You are right. It should be an easy change, do you feel okay adding it?

**Comment by abelmokadem** [comment]
Yes yes, I actually have a list of issues I want to work through, but I will do this one today. Please hold me accountable 😬
